### PR TITLE
*: update server metrics

### PR DESCRIPTION
**Pull request by shenli**
1. Use set to update connection number metrics.
2. Use hostname as instance name.
@siddontang @coocood PTAL

### CODE REVIEW

File: server/conn.go [hunk 1]
```diff
@@ -110,9 +110,9 @@ func (cc *clientConn) handshake() error {
 func (cc *clientConn) Close() error {
 	cc.server.rwlock.Lock()
 	delete(cc.server.clients, cc.connectionID)
+	connGauge.Set(float64(len(cc.server.clients)))
```
Comment (coocood) [inline server/conn.go]: assign length to a variable and set metrics outside the lock will improve performance.

File: tidb-server/main.go [hunk 2]
```diff
@@ -194,10 +194,9 @@ func pushMetric(addr string, interval time.Duration) {
 func prometheusPushClient(addr string, interval time.Duration) {
 	// TODO: TiDB do not have uniq name, so we use host+port to compose a name.
 	job := "tidb"
-	grouping := map[string]string{"instance": fmt.Sprintf("%s:%s", *host, *port)}
 	for {
 		err := push.FromGatherer(
-			job, grouping,
+			job, nil,
```
Comment (siddontang) [inline tidb-server/main.go]: my fault, we must use HostnameGroupingKey directly here.

**Comment by shenli** [comment]
@coocood @siddontang PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**